### ISSUE TITLE
Move input generation and inference for each class of models into distinct modules

### ISSUE THREAD

**Issue by hvgazula** [hassonlab/247-pickling#122]
The input generation, inference, and embeddings/logits extraction functions (as appropriate) `tfsemb_main.py` should be moved into separate scripts for `causal`, `mlm`, and `seq2seq` models.

**Comment by zkokaja** [comment]
Yes, let's flesh this out and implement a prototype for `causal` so we can implement it for other types as well.

**Comment by zkokaja** [comment]
Consider adding special tokens to causal models at the beginning, to stay true to what the model was trained on. Needs investigating.

**Comment by hvgazula** [comment]
https://datascience.stackexchange.com/questions/86566/whats-the-right-input-for-gpt-2-in-nlp

**Comment by zkokaja** [comment]
Let's talk about this again

**Comment by zkokaja** [comment]
waiting to resolve issue with whisper embedding generation replication
